packages/js/src/ui/helpers/useUncontrolledState.ts

```diff
@@ -0,0 +1,33 @@
+import { Accessor, createSignal } from 'solid-js';
+
+type UseUncontrolledState = {
+  value?: boolean;
+  fallbackValue?: boolean;
+};
+
+type UseUncontrolledStateOutput = [Accessor<boolean>, (value: boolean) => void];
```

Comment:
Do we need the output type? Can we just let it auto infer?

Reply:
it infers wrong return types due to being a conditional tuple return types.
So, we need the Output type